src/Esfa.Vacancy.Application/Queries/SearchApprenticeshipVacancies/SearchApprenticeshipVacanciesRequestValidator.cs

```diff
@@ -103,11 +103,12 @@ public SearchApprenticeshipVacanciesRequestValidator(ITrainingDetailService trai
                     .WithErrorCode(ErrorCodes.SearchApprenticeships.DistanceInMiles));
 
             RuleFor(request => request.SortBy)
-                .IsInEnum()
-                .When(request => request.SortBy.HasValue)
+                .Must(sortBy => Enum.IsDefined(typeof(SortBy), sortBy))
+                .When(request => !string.IsNullOrWhiteSpace(request.SortBy))
```

Comment:
I think since this is a value we are reading from query string, white space should also raise an error, hence we should check for null or empty.

Reply:
Updated